app/Http/Controllers/admin/TopicController.php

```diff
@@ -0,0 +1,187 @@
+        $topics = $this->topic->getAll();
+        $test_model = new Topic();
+        $fillable_columns = $test_model->getFillable();
+        foreach ($fillable_columns as $key => $value) {
+            $test_columns[$value] = $value;
+        }
+
+        return view('admin.topic.index', compact('topics', 'test_columns'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('admin.topic.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TopicFormRequest $request)
+    {
+        if (Auth::check()) {
+            $user_id = Auth::user()->id;
+            $request->merge(['user_id' => $user_id]);
+            $image = $request->file('image_topic');
+            $img = $this->saveImage($image);
+            $request->merge(['images' => $img]);
+
+            //handle select time
+            if ($request->input('set_time') == 0) {
+                $select_time = null;
+            } elseif ($request->input('set_time') == 1) {
+                $select_time = $request->input('select_time');
+
+            }
+            $request->merge(['select_time' => $select_time]);
+//            dd(Carbon::now()->hour . ':'. Carbon::now()->minute);
+            $this->topic->store($request->all());
+
+            return back()->with('success', __('message.success'));
+        }
+
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $topic = $this->topic->show($id);
+
+        return view('admin.topic.show', compact('topic'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $topic = $this->topic->find($id);
+
+        return view('admin.topic.edit', compact('topic'));
+    }
+
+    public function updateAll(TopicFormRequest $request, $id)
+    {
+        $topic = $this->topic->find($id);
+        $image = $request->file('image_topic');
+        $img = $this->saveImage($image);
+        $request->merge(['images' => $img]);
+        $topic->update($request->all());
+
+        return back()->with('success', trans('message.success'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $test = Topic::find($id);
+        $column_name = Input::get('name');
+        $column_value = Input::get('value');
+        if (Input::has('name')) {
+            $test = Topic::select()
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 137</br>Variable "column_name" is not in valid camel caps format